Files Added

Dipesh0511 opened the pull request.

Dipesh0511 committed:
Files Added

Dipesh0511 committed:
Problem 1 Added

sharmajitesh07 commented on Week-4/Problem-1 Promise.js:
```diff
@@ -0,0 +1,16 @@
+const promise = function(number){
```
I assume this question asks you to create a polyfill for promises. and not use promise itself

sharmajitesh07 commented on Week-3/Problem 1.js:
```diff
@@ -0,0 +1,47 @@
+// Memoize Function for addition of two Numbers
+var value=[];
+function add(value=[]){
+    var sum=0;
```
use let instead of var

sharmajitesh07 commented on Week-3/Problem 1.js:
```diff
@@ -0,0 +1,47 @@
+var value=[];
+function add(value=[]){
+    var sum=0;
+    for(i =0;i<value.length;i++){
```
no initialization for i ?

sharmajitesh07 commented on Week-3/Problem 1.js:
```diff
@@ -0,0 +1,47 @@
+    console.log(`Sum of both numbers is ` + sum);
+    return (sum);
+}
+function memoize(fn){
```
Lets use arrow functions where ever we can

sharmajitesh07 commented on Week-3/Problem 1.js:
```diff
@@ -0,0 +1,47 @@
+            cache.set(key,fn(...args));
+            console.log("Value Entered First Time");
+            return cache.get(key);
+        }
```
This can better simplified as 
`if(!cache.has(key)){
            cache.set(key,fn(...args));
            console.log("Value Entered First Time");
        }
       return cache.get(key);`

sharmajitesh07 commented on Week-4/Problem-2 Inheritance.js:
```diff
@@ -0,0 +1,19 @@
+    this.age = age;
+}
+
+class teacher extends person {
```
Create teacher also with function keywrod

Dipesh0511 committed:
Update Problem 1.js

Dipesh0511 committed:
New files added

Dipesh0511 committed:
Added
Merge branch 'Dipesh' of https://github.com/pesto-students/p7-jitesh-Dipesh0511 into Dipesh

Dipesh0511 committed:
Update 1.js

Dipesh0511 committed:
Update Problem 1.js
